.github/workflows/ci-pipeline.yml

```diff
@@ -0,0 +1,361 @@
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Cache Dependencies
+        uses: actions/cache@v3
+        with:
+          path: ui/node_modules
+          key: |
+            ${{ runner.os }}-modules-${{ hashFiles('ui/yarn.lock') }}
+          restore-keys: ${{ runner.os }}-modules-
+
+      - name: Install
+        working-directory: ui
+        run: yarn install
+
+      - name: Lint code
+        run: make lint-ui
+
+      - name: Build UI
+        env:
+          NODE_OPTIONS: "--max_old_space_size=4096"
+        run: make build-ui
+
+      - name: Publish Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: mlp-ui-dist
+          path: ui/build/
+          retention-days: ${{ env.ARTIFACT_RETENTION_DAYS }}
+
+  build-api:
+    runs-on: ubuntu-latest
+    outputs:
+      api-version: ${{ steps.build-image.outputs.api-version }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Build Docker image
+        id: build-image
+        run: |
+          set -o pipefail
+          make build-api-image | tee output.log
+          echo "api-version=$(sed -n 's%version: \(.*\)%\1%p' output.log)" >> $GITHUB_OUTPUT
+
+      - name: Save Docker image
+        run: |
+          docker image save \
+            --output mlp-api.${{ steps.build-image.outputs.api-version }}.tar \
+            mlp-api:${{ steps.build-image.outputs.api-version }}
+
+      - name: Publish Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: mlp-api.${{ steps.build-image.outputs.api-version }}.tar
+          path: mlp-api.${{ steps.build-image.outputs.api-version }}.tar
+          retention-days: ${{ env.ARTIFACT_RETENTION_DAYS }}
+
+  test-api:
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: ${{ github.workspace }}/.go
+    services:
+      postgres:
+        image: postgres:12.4
+        env:
+          POSTGRES_DB: mlp
+          POSTGRES_USER: mlp
+          POSTGRES_PASSWORD: mlp
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+      keto:
+        image: oryd/keto:v0.4.5
+        env:
+          DSN: memory
+        ports:
+          - 4466:4466
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Cache Dependencies
+        uses: actions/cache@v3
+        with:
+          path: .go/pkg/mod/
+          key: |
+            gomod-${{ hashFiles('go.mod') }}
+          restore-keys: gomod-
+
+      - name: Run Integration Test
+        env:
+          DATABASE_HOST: localhost
+          DATABASE_NAME: mlp
+          DATABASE_USER: mlp
+          DATABASE_PASSWORD: mlp
+          KETO_URL: http://localhost:4466
+        run: make it-test-api-ci
+
+      - name: Lint code
```

Comment:
Should we run lint before integration test?

Reply:
What's your motivation for this order? Considering that the execution time of the `it-test` and `lint` is ~ the same, I don't see a big difference in what runs first:
<img width="1294" alt="Screenshot 2022-10-31 at 7 39 56 PM" src="https://user-images.githubusercontent.com/1886194/198999711-ef8eff76-4e4f-4090-ac18-dabd8648dd0b.png">

Reply:
For our experience in Merlin, test-api takes longer time to finish than lint-api job [1]. However, the current MLP tests are still more straightforward and fewer that's why the test and lint duration Is similar. But in the longer time, if we have more tests, I believe the test duration would be longer. 

[1] Merlin jobs:
<img width="181" alt="image" src="https://user-images.githubusercontent.com/8122852/199001184-529315b5-b6e4-44e6-bb02-1b9bee689b4c.png">

Reply:
Another option is to separate lint and test into two jobs so they can run in parallel as well.

Reply:
I put lint before the integration test, this should be good for now